libs/universal-dapp.js

```diff
@@ -0,0 +1,333 @@
+        .append( $('<div class="call"/>').text('Call') )
+
+    this.$el.append( $('<div class="poweredBy" />')
+        .html("<a href='http://github.com/d11e9/universal-dapp'>Universal ÐApp</a> powered by The Blockchain") )
+
+    this.$el.append( $legend )
+    return this.$el;
+}
+
+UniversalDApp.prototype.getABIInputForm = function (cb){
+    var self = this;
+    var $el = $('<div class="udapp-setup" />');
+    var $jsonInput = $('<textarea class="json" placeholder=\'[ { "name": name, "bytecode": bytyecode, "interface": abi }, { ... } ]\'/>')
+    var $createButton = $('<button class="udapp-create"/>').text('Create a Universal ÐApp')
+    $createButton.click(function(ev){
+        var contracts =  $.parseJSON( $jsonInput.val() );
+        if (cb) {
+            var err = null;
+            var dapp = null;
+            try {
+                dapp = new UniversalDApp( contracts, self.options );
+            } catch(e) {
+                err = e;
+            }
+            cb( err, dapp )
+        } else {
+            self.contracts = contracts;
+            self.$el.empty().append( self.render() )
+        }
+    })
+    $el.append( $jsonInput ).append( $createButton )
+    return $el;
+}
+
+
+UniversalDApp.prototype.getCreateInterface = function ($container, contract) {
+    var self = this;
+    var $createInterface = $('<div class="create"/>');
+    if (this.options.removable) {
+        var $close = $('<div class="udapp-close" />')
+        $close.click( function(){ self.$el.remove(); } )
+        $createInterface.append( $close );
+    }
+    var $newButton = this.getInstanceInterface( contract )
+    var $atButton = $('<button class="atAddress"/>').text('At Address').click( function(){ self.clickContractAt( self, $container, contract ) } );
+    $createInterface.append( $atButton ).append( $newButton );
+    return $createInterface;
+}
+
+UniversalDApp.prototype.getInstanceInterface = function (contract, address, $target) {
+    var self = this;
+    var abi = JSON.parse(contract.interface).sort(function(a,b){
+        if (a.name > b.name) return -1;
+        else return 1;
+    }).sort(function(a,b){
+        if (a.constant == true) return -1;
+        else return 1;
+    });
+    var funABI = this.getConstructorInterface(abi);
+    var $createInterface = $('<div class="createContract"/>');
+
+    var appendFunctions = function (address, $el){
+        
+        var $instance = $('<div class="instance"/>');
+        if (self.options.removable_instances) {
+            var $close = $('<div class="udapp-close" />')
+            $close.click( function(){ $instance.remove(); } )
+            $instance.append( $close );
+        }
+        var $title = $('<span class="title"/>').text( contract.name + " at " + (self.options.vm ? '0x' : '') + address.toString('hex') );
+        $title.click(function(){
+            $instance.toggleClass('hide');
+        });
+
+        $events = $('<div class="events"/>');
+        if (!self.options.vm){
```

Comment:
Requires upgrading `ethereumjs-vm` and since we are already upgrading `web3` in this PR I propose deferring adding this for another PR. /cc @chriseth